### DISCUSSION TITLE
New ATSs/ETSs for the integration of Good Practice "Data and Service Linking Simplification"

### DISCUSSION THREAD

**Discussion by fabiovinci** [INSPIRE-MIF/helpdesk-validator#1009] [Announcements]
Dear Community,

this message is to inform you that the ATSs/ETSs for the integration of the [Good Practice "Data and Service Linking Simplification"](https://github.com/INSPIRE-MIF/gp-data-service-linking-simplification) into the Reference Validator are under development and the related tests will be soon available.

Good Practice includes two main parts:

- Part A ([Section 8](https://github.com/INSPIRE-MIF/gp-data-service-linking-simplification/blob/main/good-practice/data-service-linking-simplification-spec.md#8-part-a-data-service-linking-simplification-requirements-classes-)): "Data-Service Linking Simplification. Requirements classes", which includes requirements/recommendations to be added/updated in the relevant TGs, and in particular:
  - Section 8.1: INSPIRE Data Set Metadata Resource Locator, which requires:
    - Change in the MD TG (Issue [#108](https://github.com/INSPIRE-MIF/technical-guidelines/issues/108))
    - New ATS/ETS: “Conformance Class 8: INSPIRE data sets and data set series linked service metadata”
  - Section 8.2: INSPIRE Network Service Metadata Coupled Resource, which requires:
    - Change in the MD TG (Issue [#108](https://github.com/INSPIRE-MIF/technical-guidelines/issues/108)) 
    - Change in the Service TGs (Issues [#105](https://github.com/INSPIRE-MIF/technical-guidelines/issues/105), [#110](https://github.com/INSPIRE-MIF/technical-guidelines/issues/110) and [#112](https://github.com/INSPIRE-MIF/technical-guidelines/issues/112))
-  Part B ([Section 9](https://github.com/INSPIRE-MIF/gp-data-service-linking-simplification/blob/main/good-practice/data-service-linking-simplification-spec.md#9-part-b-use-of-inspire-conformant-standard-capabilities-documents-)): Use of INSPIRE conformant standard capabilities documents, which requires:
   - Update of the ATS/ETS for View Service ([Updated ATS](https://github.com/inspire-eu-validation/view-service))
   - Update of the ATS/ETS for Download Service - WFS ([Updated ATS](https://github.com/inspire-eu-validation/download-wfs))
   - Update of the ATS/ETS for Download Service - ATOM ([Updated ATS](https://github.com/inspire-eu-validation/download-atom))

The new ATS “[Conformance Class 8: INSPIRE data sets and data set series linked service metadata](https://github.com/inspire-eu-validation/metadata/tree/2.0/ds-linked-service)” is already available, so any comment/suggestion is more than welcome.

Status updates will follow in this discussion.

**Comment by fabiovinci** [comment]
Dear Community,

we inform you that the “**Conformance Class 8: INSPIRE data sets and data set series linked service metadata**” is available on the [staging instance](http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/test-selection/index.html) and it is ready for testing.

You will find it, as preselected, in the "Advanced options" menu.

Feel free to test it and report any problem/suggestion/comment through the issue tracker.

![image](https://github.com/INSPIRE-MIF/helpdesk-validator/assets/22212406/97c688f7-4656-44cf-afb4-fe2c9ec47c08)

**Reply by hogredan** [reply]
Thank you for the opportunity to test the new Conformance Class 8. 

However, this new Conformance Class should no longer be preselected when publishing it in the production environment, as the requirements from the good practice do not have to be fulfilled by all data sets metadata.

**Reply by fabiovinci** [reply]
Dear @hogredan,

it was decided to set the new Conformance Class 8 selected by default, to “_force_” the users to provide and encode the Resource Locator elements as requested by the Good Practice for different reasons:
- the multiplicity of this element is zero or more, but it is "_mandatory if a URL is available to obtain more information on the resource, and/or access related services._"
- the INSPIRE legal framework requires that data sets are made available through View and Download services, which in turn implies that at least two resource locator elements need to be expressed in the data set metadata: one for a View Service and one for a Download Service.
- setting up the correct resource locators is important for the connection between the data set and the services that provide access to it (e.g. it facilitates the machine-to-machine element recognition made by an INSPIRE software implementation such as the INSPIRE Geoportal).

In any case, this is our proposal and may be subject to review.

**Reply by hogredan** [reply]
Dear @fabiovinci ,

this would break existing implementations with well-working data-service coupling and would mean (if I am right) that CC 8 is also used to check the conformity of dataset metadata in the INSPIRE monitoring. 

And this was not agreed in the MIG-T. The simplified data-service coupling is only an alternative approach and to be used in parallel with the current one (as described in the TG documents on metadata, view services and download services). Thus, the current approach must remain valid!

Therefore, in our view, the new CC should not be selected by default.

**Reply by fabiovinci** [reply]
Dear @hogredan,

there are no issues with the existing implementations as CC8 can always be deselected and was not used in the monitoring.

The current approach remains valid and compliance with CC8 is only a "recommendation" for users who are not implementing the "Data and Service Linking Simplification" good practice.

**Comment by fabiovinci** [comment]
Dear Community,

we inform you that the sample metadata "[Dataset_metadata_2.0_example.xml](https://github.com/INSPIRE-MIF/helpdesk-validator/blob/master/examples/Dataset_metadata_2.0_example.xml)" was updated to be compliant with Conformance Class 8.

It now contains the two "Resource Locator" elements encoded according to the [Good Practice "Data and Service Linking Simplification"](https://github.com/INSPIRE-MIF/gp-data-service-linking-simplification).

**Comment by fabiovinci** [comment]
Dear Community,

we inform you that [Part A. Data-Service Linking Simplification. Requirements classes](https://github.com/INSPIRE-MIF/gp-data-service-linking-simplification/blob/main/good-practice/data-service-linking-simplification-spec.md#8-part-a-data-service-linking-simplification-requirements-classes-) of the Good Practice was implemented.

All the required TG changes were included in the latest [TG release](https://github.com/INSPIRE-MIF/technical-guidelines/releases/tag/v2024.1) and the new Conformance Class 8 is available in the production instance of the Reference Validator.

**Comment by fabiovinci** [comment]
Dear Community,

we inform you that we are working on the implementation of [Part B. Use of INSPIRE conformant standard capabilities documents](https://github.com/INSPIRE-MIF/gp-data-service-linking-simplification/blob/main/good-practice/data-service-linking-simplification-spec.md#9-part-b-use-of-inspire-conformant-standard-capabilities-documents-), and that we already implemented the requested changes in the **ATS/ETS for View Service**.

The updated ATS is available [here](https://github.com/inspire-eu-validation/view-service/tree/master/iso-19128). 
The related ETS has been updated accordingly and is now ready for testing in the [staging instance](https://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/home/index.html).

Feel free to test both ATS and ETS and report any problem/suggestion/comment through the [helpdesk-validator issue tracker](https://github.com/INSPIRE-MIF/helpdesk-validator/issues)..

**Comment by fabiovinci** [comment]
Dear Community,

we inform you that we are working on implementing the requested changes in **ATS/ETS for Download and ATOM Services**.

The updated ATS for WFS is available [here](https://github.com/inspire-eu-validation/download-wfs/tree/ds_link_simpl/wfs-pre-defined). 
The updated ATS for ATOM is available [here](https://github.com/inspire-eu-validation/download-atom/tree/ds_link_simpl/atom-pre-defined). 

The related ETSs are under finalization and will be available in the next release of the Validator.

Please feel free to test the ATSs and report any problem/suggestion/comment through the [helpdesk-validator issue tracker](https://github.com/INSPIRE-MIF/helpdesk-validator/issues).

**Comment by fabiovinci** [comment]
Dear Community,

we just wanted to let you know that the updated tests for WFS and ATOM are ready for testing in the [staging instance](https://inspire.ec.europa.eu/validator-staging/home/index.html).

**Comment by fabiovinci** [comment]
Dear Community,

we would like to inform you that the integration of the [Good Practice "Data and Service Linking Simplification"](https://github.com/INSPIRE-MIF/gp-data-service-linking-simplification) into the Reference Validator has been completed.

All the related tests are now available in the [production instance](https://inspire.ec.europa.eu/validator/home/index.html).
